src/index.ts

```diff
@@ -7,31 +7,44 @@ import type {
 import { HeaderMap } from '@apollo/server';
 import type { WithRequired } from '@apollo/utils.withrequired';
 import type {
+  ALBEvent,
+  ALBResult,
   APIGatewayProxyEvent,
-  APIGatewayProxyEventHeaders,
-  APIGatewayProxyEventQueryStringParameters,
   APIGatewayProxyEventV2,
   APIGatewayProxyResult,
   APIGatewayProxyStructuredResultV2,
   Context,
   Handler,
 } from 'aws-lambda';
 
-export type GatewayEvent = APIGatewayProxyEvent | APIGatewayProxyEventV2;
+export type IncomingEvent =
+  | APIGatewayProxyEvent
+  | APIGatewayProxyEventV2
+  | ALBEvent;
+
+/**
+ * @deprecated Use {IncomingEvent} instead
+ */
+export type GatewayEvent = IncomingEvent;
 
 export interface LambdaContextFunctionArgument {
-  event: GatewayEvent;
+  event: IncomingEvent;
   context: Context;
 }
 
 export interface LambdaHandlerOptions<TContext extends BaseContext> {
   context?: ContextFunction<[LambdaContextFunctionArgument], TContext>;
 }
 
-type LambdaHandler = Handler<
-  GatewayEvent,
-  APIGatewayProxyStructuredResultV2 | APIGatewayProxyResult
->;
+export type HandlerResult = (
+  | APIGatewayProxyStructuredResultV2
+  | APIGatewayProxyResult
+  | ALBResult
+) & {
+  statusCode: number;
```

Comment:
Is it ok to assume this? I don't actually know, but the types for `APIGatewayProxyStructuredResultV2` include

```ts
statusCode?: number | undefined;
```

Are the types lying? (maybe we can submit a PR to correct them if so?)